src/quick/items/qquickitemview.cpp

```diff
@@ -1512,7 +1512,7 @@ QQuickItemViewPrivate::~QQuickItemViewPrivate()
 
 bool QQuickItemViewPrivate::isValid() const
 {
-    return model && model->count() && model->isValid();
+    return model && ((model->count() && model->isValid()) || !visibleItems.isEmpty());
```

Comment:
The alternative is to leave isValid() as-is and modify the isValid() test in setContentSize() and anywhere else required to be this expanded version of isValid().  Which is not a good solution for maintainability, but might be preferred in this case?